lib/APPSensorFusion/App.cpp

```diff
@@ -164,6 +166,19 @@ void App::sendSensorData() const
     payload.magnetometerValueY  = magnetometerValues.valueY;
     payload.turnRate            = turnRates.valueZ;
 
+    uint32_t duration = 0U;
+    if (true == m_firstIteration)
+    {
+        duration         = SEND_SENSOR_DATA_PERIOD;
+        m_firstIteration = false;
+    }
+    duration = m_measurementTimer.getCurrentDuration();
+    m_measurementTimer.restart();

```

Comment:
Restart the timer at the end of the function. Otherwise the timer is already running and you have not started to do real work yet